java/ql/test/library-tests/dataflow/entrypoint-types/EntryPointTypesTest.java

```diff
@@ -0,0 +1,34 @@
+public class EntryPointTypesTest {
+
+    static class TestObject {
+        public String field1;
+        private String field2;
+        private AnotherTestObject field3;
+
+        public String getField2() {
+            return field2;
+        }
+
+        public AnotherTestObject getField3() {
+            return field3;
+        }
+    }
+
+    static class AnotherTestObject {
+        public String field4;
+        private String field5;
+
+        public String getField5() {
+            return field5;
+        }
+    }
+
+    private static void sink(String sink) {}
+
+    public static void test(TestObject source) {
```

Comment:
Could you add some test cases for when `TestObject` is a parameterized type?

Reply:
Sure, added in 8250cec